pkg/network/common/dns.go

```diff
@@ -154,8 +154,13 @@ func (d *DNS) getIPsAndMinTTL(domain string) ([]net.IP, time.Duration, error) {
 				case *dns.A:
 					ips = append(ips, t.A)
 
-					if minTTL == 0 || t.Hdr.Ttl < minTTL {
-						minTTL = t.Hdr.Ttl
+					// We could potentially get zero TTL as we try to query the DNS
+					// server at the end of the TTL value for the domain.
+					// Adding 1 sec will ensure non zero minTTL
+					ttl := t.Hdr.Ttl + 1
```

Comment:
Can we make minTTL an int32 (what does dns spec the ttl as?)  Then we can set the minTTL to -1 when we initialize it.

Then we can just  compare the ttl against the minTTL and not have to do the 1s offset.

Reply:
github.com/miekg/dns has t.Hdr.Ttl as uint32 which is as per the DNS spec. So converting to int32 will be an issue (not sure whether so high TTL values are set in real world or not)

Reply:
Ok, sounds good to me.